File: app/src/main/java/com/kakacat/minitool/bingpic/Contract.java
package com.kakacat.minitool.bingpic;

import com.facebook.drawee.view.SimpleDraweeView;
import com.kakacat.minitool.common.base.IPresenter;
import com.kakacat.minitool.common.base.IView;

import java.util.List;

public interface Contract {

    interface View extends IView<Presenter> {
        void showBigImage(android.view.View view);

        void showOptionDialog(android.view.View view);

        void saveImage();

        void onUpdateImagesCallBack();

        void onSaveImageCallBack(String result);
    }

    interface Presenter extends IPresenter {
        void initData();

        void loadMore();

        void saveImage(SimpleDraweeView imageView);

        List<String> getAddressList();
    }
}

File: app/src/main/java/com/kakacat/minitool/bingpic/Contract.kt
package com.kakacat.minitool.bingpic

import com.facebook.drawee.view.SimpleDraweeView
import com.kakacat.minitool.common.base.IPresenter
import com.kakacat.minitool.common.base.IView

interface Contract {
    interface View : IView<Presenter?> {
        fun showBigImage(view: android.view.View?)
        fun showOptionDialog(view: android.view.View?)
        fun saveImage()
        fun onUpdateImagesCallBack()
        fun onSaveImageCallBack(result: String?)
    }

    interface Presenter : IPresenter {
        fun loadMore()
        fun saveImage(imageView: SimpleDraweeView?)
        val addressList: List<String?>?
    }
}